zheng-common/src/test/java/com/zheng/common/util/PaginatorTest.java

```diff
@@ -0,0 +1,116 @@
+package com.zheng.common.util;
+
+import com.diffblue.deeptestutils.mock.DTUMemberMatcher;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.lang.reflect.Method;
+import javax.servlet.http.HttpServletRequest;
+
+@RunWith(PowerMockRunner.class)
+public class PaginatorTest {
+
+  @PrepareForTest({Paginator.class, HttpServletRequest.class})
+  @Test
+  public void constructor() throws Exception {
+
+    // Arrange
+    final long total = 0L;
+    final int page = 0;
+    final int rows = 1;
+    final HttpServletRequest request = PowerMockito
+        .mock(HttpServletRequest.class);
+    final Method getRequestURIMethod =
+        DTUMemberMatcher.method(
+            HttpServletRequest.class, "getRequestURI");
+    PowerMockito.doReturn("BAZ").when(request, getRequestURIMethod)
+        .withNoArguments();
+    final Method getQueryStringMethod =
+        DTUMemberMatcher.method(
+            HttpServletRequest.class, "getQueryString");
+    PowerMockito.doReturn("1a 2b 3c").when(
+        request, getQueryStringMethod)
+        .withNoArguments();
+
+    // Act, creating object to test constructor
+    final Paginator objectUnderTest = new Paginator(total, page, rows, request);
+
+    // Assert side effects
+    Assert.assertEquals(5, objectUnderTest.getStep());
+    Assert.assertEquals("BAZ", objectUnderTest.getUrl());
+    Assert.assertEquals("1a 2b 3c", objectUnderTest.getQuery());
+    Assert.assertEquals("page", objectUnderTest.getParam());
+    Assert.assertEquals(1, objectUnderTest.getPage());
+    Assert.assertEquals(1, objectUnderTest.getRows());
+  }
+
+  @Test
+  public void setTotal() {
```

Comment:
We don't submit tests for getters & setters.

Reply:
This one has interesting side-effects that ought to be tested.

Reply:
@allredj Run it past @peterschrammel

Reply:
LGTM